test/functions_linearalgebra.jl

```diff
@@ -0,0 +1,67 @@
+using LinearAlgebra
+using NamedDims
+using NamedDims: names
+using Test
+
+# LinearAlgebra
+@testset "lu" begin
+    nda = NamedDimsArray{(:foo, :bar)}([1.0 2; 3 4])
+    x = lu(nda)
+    @test names(x.L) == (:foo, :_)
+    @test names(x.U) == (:_, :bar)
+    @test names(x.p) == (:foo,)
+    @test names(x.P) == (:foo, :foo)
+
+    # Idenity opperations should give back original names
+    @test names(x.P * nda) == (:foo, :bar)
+    @test names(x.L * x.U) == (:foo, :bar)
+    @test names(nda[x.p, :]) == (:foo, :bar)
+end
+
+@testset "lq" begin
+    nda = NamedDimsArray{(:foo, :bar)}([1.0 2; 3 4])
+    x = lq(nda)
+    @test names(x.L) == (:foo, :_)
+    @test names(x.Q) == (:_, :bar)
+
+    # Idenity opperations should give back original names
+    @test names(x.L * x.Q) == (:foo, :bar)
+end
+
+@testset "svd" begin
+    nda = NamedDimsArray{(:foo, :bar)}([1.0 2; 3 4])
+    x = svd(nda)
+    @test names(x.U) == (:foo, :_)
+    @test names(x.S) == (:_,)
+    @test names(x.V) == (:_, :bar)
+    @test names(x.Vt) == (:bar, :_)
+
+    # Identity operation should give back original nam,es
+    @test names(x.U * Diagonal(x.S) * x.V) == (:foo, :bar)
```

Comment:
isn't the input recovered by the operation `U * Diagonal(S) * Vt`? i.e. the names of `V` and `Vt` are flipped in this PR?

Reply:
Yes, I believe the names should be on the first dimension for `U` and `V` based on this visualization.

https://en.wikipedia.org/wiki/File:Singular_value_decomposition_visualisation.svg